lib/aws/request.js

```diff
@@ -30,7 +31,7 @@ const proxy =
 
 const proxyOptions = {};
 if (proxy) {
-  Object.assign(proxyOptions, new URL(proxy));
+  Object.assign(proxyOptions, url.parse(proxy));
```

Comment:
I think i'tll be nice to add a comment indicating why we're using `url.parse` instead of `URL`, also linking related issue in `https-proxy-agent`

Reply:
comments added